lib/dialects/sqlite/query-generator.js

```diff
@@ -257,8 +257,11 @@ class SQLiteQueryGenerator extends MySqlQueryGenerator {
     return { query, bind };
   }
 
-  truncateTableQuery(tableName) {
-    return `DELETE FROM ${this.quoteTable(tableName)}`;
+  truncateTableQuery(tableName, options = {}) {
+    return [
+      `DELETE FROM ${this.quoteTable(tableName)}`,
+      options.restartIdentity ? `; DELETE FROM ${this.quoteTable('sqlite_sequence')} WHERE ${this.quoteIdentifier('name')} = '${tableName.schema}.${tableName.table}';` : ''
```

Comment:
```suggestion
      options.restartIdentity ? `; DELETE FROM ${this.quoteTable('sqlite_sequence')} WHERE ${this.quoteIdentifier('name')} = '${this.escape(tableName.schema)}.${this.escape(tableName.table)}';` : ''
```

Reply:
Oh hang on, in both my original and this suggestion we don't handle cases without schema. We need the logic from quoteTable.

Reply:
You can use ```Utils.addTicks(Utils.removeTicks(this.quoteTable(table), '`'), "'");```